src/mqt/ionshuttler/linear/dd/global_dd.py

```
@@ -0,0 +1,335 @@
+    def from_dict(cls, data: object) -> GlobalDDReport:
+        """Restore a global-DD report from JSON-compatible values.
+
+        Returns:
+            The restored global-DD report.
+
+        Raises:
+            ValueError: If the serialized report is malformed.
+        """
+        if not isinstance(data, dict):
+            msg = "global DD report must be a JSON object"
+            raise ValueError(msg)  # ruff: ignore[type-check-without-type-error] - Malformed JSON uses ValueError.
```

Ruff (0.16.1)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)